git-cliff-core/src/github.rs

```diff
@@ -0,0 +1,261 @@
+}
+
+/// Representation of a single commit.
+#[derive(Default, Debug, Clone, PartialEq, Serialize, Deserialize)]
+pub struct GitHubCommit {
+	/// SHA.
+	pub sha:    String,
+	/// Author of the commit.
+	pub author: Option<GitHubCommitAuthor>,
+}
+
+impl GitHubEntry for GitHubCommit {
+	fn url(owner: &str, repo: &str, page: i32) -> String {
+		format!(
+			"{GITHUB_API_URL}/repos/{}/{}/commits?per_page={MAX_PAGE_SIZE}&\
+			 page={page}",
+			owner, repo
+		)
+	}
+	fn buffer_size() -> usize {
+		10
+	}
+}
+
+/// Author of the commit.
+#[derive(Default, Debug, Clone, PartialEq, Serialize, Deserialize)]
+pub struct GitHubCommitAuthor {
+	/// Username.
+	pub login: Option<String>,
+}
+
+/// Representation of a single pull request.
+#[derive(Default, Debug, Clone, PartialEq, Serialize, Deserialize)]
+pub struct GitHubPullRequest {
+	/// Pull request number.
+	pub number:           i64,
+	/// SHA of the merge commit.
+	pub merge_commit_sha: Option<String>,
+}
+
+impl GitHubEntry for GitHubPullRequest {
+	fn url(owner: &str, repo: &str, page: i32) -> String {
+		format!(
+			"{GITHUB_API_URL}/repos/{}/{}/pulls?per_page={MAX_PAGE_SIZE}&\
+			 page={page}&state=closed",
+			owner, repo
+		)
+	}
+
+	fn buffer_size() -> usize {
+		5
+	}
+}
+
+/// Metadata of a GitHub release.
+#[derive(Debug, Default, Clone, Eq, PartialEq, Deserialize, Serialize)]
+pub struct GitHubReleaseMetadata {
+	/// Contributors.
+	pub contributors: Vec<GitHubContributor>,
+}
+
+/// Representation of a GitHub contributor.
+#[derive(Debug, Default, Clone, Eq, Deserialize, Serialize)]
+pub struct GitHubContributor {
+	/// Username.
+	pub username:      Option<String>,
+	/// The pull request that the user created.
+	pub pr_number:     Option<i64>,
+	/// Whether if the user contributed for the first time.
+	pub is_first_time: bool,
+}
+
+impl PartialEq for GitHubContributor {
+	fn eq(&self, other: &Self) -> bool {
+		self.username.eq(&other.username)
+	}
+}
+
+impl Hash for GitHubContributor {
+	fn hash<H: Hasher>(&self, state: &mut H) {
+		self.username.hash(state);
+	}
+}
+
+/// HTTP client for handling GitHub REST API requests.
+#[derive(Debug, Clone)]
+pub struct GitHubClient {
+	/// Owner of the repository.
+	owner:  String,
+	/// GitHub repository.
+	repo:   String,
+	/// HTTP client.
+	client: ClientWithMiddleware,
+}
+
+/// Constructs a GitHub client from the remote configuration.
+impl TryFrom<Remote> for GitHubClient {
+	type Error = Error;
+	fn try_from(remote: Remote) -> Result<Self> {
+		if !remote.is_set() {
+			return Err(Error::RemoteNotSetError);
+		}
+		let mut headers = HeaderMap::new();
+		headers.insert(
+			reqwest::header::ACCEPT,
+			HeaderValue::from_static("application/vnd.github+json"),
+		);
+		if let Some(token) = remote.token {
+			headers.insert(
+				reqwest::header::AUTHORIZATION,
+				format!("Bearer {}", token.expose_secret()).parse()?,
+			);
+		}
+		headers.insert(reqwest::header::USER_AGENT, USER_AGENT.parse()?);
+		let client = Client::builder()
```

Comment:
It is possible to keep the TCP connection open using https://docs.rs/reqwest/latest/reqwest/struct.ClientBuilder.html#method.tcp_keepalive, so that might 

Normally it is 60 seconds by default, but it is [disabled](https://docs.rs/reqwest/latest/src/reqwest/async_impl/client.rs.html#186) at the moment.